lib/Controller/User.php

```diff
@@ -861,8 +862,17 @@ public function delete($userId)
     {
         $user = $this->userFactory->getById($userId);
 
-        if (!$this->getUser()->checkDeleteable($user))
+        if (!$this->getUser()->checkDeleteable($user)) {
             throw new AccessDeniedException();
+        }
+
+        if ($this->getUser()->userId === $userId) {
+            throw new \InvalidArgumentException(__('Cannot delete your own User from the CMS.'));
```

Comment:
Is this the right type of InvalidArgumentException? Don't we use the support library?